clang/include/clang/Basic/LangOptions.h

```diff
@@ -706,6 +706,10 @@ class LangOptions : public LangOptionsBase {
     return OpenCL || CUDA;
   }
 
+  /// Return true if the dispatch size for an offload language only uses one
+  /// dimension.
+  bool gridSizeIsOneDimension() const { return CUDA || HIP || OpenMP; }
```

Comment:
CUDA and HIP don't support 3d grid? Did I misunderstand?

Reply:
https://docs.nvidia.com/cuda/cuda-c-programming-guide/#execution-configuration Isn't it a `dim3`?